src/sage/combinat/bijectionist.py

```diff
@@ -0,0 +1,3230 @@
+# -*- coding: utf-8 -*-
+r"""
+A bijectionist's toolkit
+
+AUTHORS:
+
+- Alexander Grosz, Tobias Kietreiber, Stephan Pfannerer and Martin
+  Rubey (2020-2022): Initial version
+
+Quick reference
+===============
+
+.. csv-table::
+    :class: contentstable
+    :widths: 30, 70
+    :delim: |
+
+    :meth:`~Bijectionist.set_statistics` | Declare statistics that are preserved by the bijection.
+    :meth:`~Bijectionist.set_value_restrictions` | Restrict the values of the statistic on an element.
+    :meth:`~Bijectionist.set_constant_blocks` | Declare that the statistic is constant on some sets.
+    :meth:`~Bijectionist.set_distributions` | Restrict the distribution of values of the statistic on some elements.
+    :meth:`~Bijectionist.set_intertwining_relations` | Declare that the statistic intertwines with other maps.
+    :meth:`~Bijectionist.set_quadratic_relation` | Declare that the statistic satisfies a certain relation.
+    :meth:`~Bijectionist.set_homomesic` | Declare that the statistic is homomesic with respect to a given set partition.
+
+
+    :meth:`~Bijectionist.statistics_table` | Print a table collecting information on the given statistics.
+    :meth:`~Bijectionist.statistics_fibers` | Collect elements with the same statistics.
+
+    :meth:`~Bijectionist.constant_blocks` | Return the blocks on which the statistic is constant.
+    :meth:`~Bijectionist.solutions_iterator` | Iterate over all possible solutions.
+    :meth:`~Bijectionist.possible_values` | Return all possible values for a given element.
+    :meth:`~Bijectionist.minimal_subdistributions_iterator` | Iterate over the minimal subdistributions.
+    :meth:`~Bijectionist.minimal_subdistributions_blocks_iterator` | Iterate over the minimal subdistributions.
+
+A guided tour
+=============
+
+    EXAMPLES:
+
+    We find a statistic `s` such that
+    `(s, wex, fix) \sim (llis, des, adj)`::
+
+        sage: N = 3
+        sage: A = B = [pi for n in range(N+1) for pi in Permutations(n)]
+        sage: alpha1 = lambda p: len(p.weak_excedences())
```

Comment:
```suggestion
        sage: def alpha1(p): return len(p.weak_excedences())
```
and similar here to be more pythonic.

Reply:
I admit I dislike this one, it is longer and harder to read for me.  Are you sure?

Reply:
Yes, although the "proper" way would be
```
sage: def alpha1(p):
....:     return len(p.weak_excedences())
```